x-pack/plugins/ml/server/saved_objects/checks.ts

```diff
@@ -263,15 +272,21 @@ export function checksFactory(
     };
   }
 
-  async function canDeleteJobs(
+  async function canDeleteMLSpaceAwareItems(
     request: KibanaRequest,
-    jobType: JobType,
-    jobIds: string[],
+    jobType: JobType | TrainedModelType,
```

Comment:
as I mentioned above, worth creating a union type `MLSpaceAwareItem` because it's been used in multiple places